eng/dependabot/Packages.props

```diff
@@ -10,47 +10,44 @@
   these properties to override package versions if necessary. -->
 
   <ItemGroup>
-    <PackageReference Update="BenchmarkDotNet" Version="0.13.1" />
-    <PackageReference Update="BenchmarkDotNet" Condition="'$(BenchmarkDotNetVersion)' != ''" Version="$(BenchmarkDotNetVersion)" />
+    <PackageVersion Include="BenchmarkDotNet" Version="0.13.1" />
+    <PackageVersion Update="BenchmarkDotNet" Condition="'$(BenchmarkDotNetVersion)' != ''" Version="$(BenchmarkDotNetVersion)" />
```

Comment:
Is it a valid use or should `VersionOverride` be used here?
https://devblogs.microsoft.com/nuget/introducing-central-package-management/#overriding-package-versions

Reply:
Good question! This is an odd usage, but I think it's correct.

These overrides generally do not apply--in local and "normal" official builds, there is no `$(BenchmarkDotNetVersion)` set, so the conditions don't apply and only the first `PackageVersion` is relevant.

However, during source-build scenarios, there's a need to specify dependencies to an exact version that matches all over the .NET SDK product (not just MSBuild). In those builds, our repo is invoked with a bunch of global properties to override our versions to specific ones, so the second lines in this file apply.

I think this is distinct from a central-package-versioning `VersionOverride` because we want the sourcebuild versions to affect the whole repo, while `VersionOverride` is for "this project within the repo is special and needs its own version".